fix(ins): update routed ip docs (#2536)
scaleway/docs-content: compute/instances/concepts.mdx

## IP mobility

> The legacy public network is provided through a network component managing a carrier-grade NAT with NAT IP addresses.

Reviewer: I feel there are too many `NAT`.
Maybe we can rephrase this way:
```suggestion
The legacy public network is provided through carrier-grade NAT infrastructure.
```

When an Instance is started, it is assigned a NAT IP address that is associated with the underlying node hosting the Instance.
The carrier-grade NAT establishes a one-to-one mapping between the public IP and the NAT IP assigned to the Instance during provisioning. Consequently, if the underlying node changes, the NAT IP address will also change.

> With IP mobility, there are no NAT IPs anymore. The public IP is entirely routed to the Instance. The main benefit is that your underlying IP no longer changes when you stop and start your Instance or if it gets provisioned on another hypervisor. But this also means that if you do not attach a public IP address to your Instance, then it will not have any network access.

Reviewer: This still lack a bit of explanation between Flexible and Dynamic IP addresses. You are only assured to keep your IP address upon stop/start if and only if your have attached a Flexible public IP address to your instances.
For those relying on a Dynamic IP address, the behavior remains the same whether you have a routed IP enabled instance or a _legacy_ one.

## IP mobility

The legacy public network is provided through carrier-grade NAT infrastructure.
When an Instance is started, it is assigned a NAT IP address that is associated with the underlying node hosting the Instance.
The carrier-grade NAT establishes a one-to-one mapping between the public IP and the NAT IP assigned to the Instance during provisioning. Consequently, if the underlying node changes, the NAT IP address will also change.

> With IP mobility, there are no NAT IPs anymore. The public IP is entirely routed to the Instance. The main benefit is that your underlying IP no longer changes when you stop and start your Instance or if it gets provisioned on another hypervisor. But this also means that if you do not attach a public IP address to your Instance, then it will not have any network access.

Reviewer 2: ```suggestion
With IP mobility, NAT IPs are no longer available. The public IP is entirely routed to the Instance. The main benefit is that your underlying IP no longer changes when you stop and start your Instance or if it gets provisioned on another hypervisor. But this also means that if you do not attach a public IP address to your Instance, then it will not have any network access.
```

<Message type="note">

>   Your IP address will be retained during stop/start operations only if you have linked a flexible public IP address to your Instance. If you are using a dynamic IP address, the behavior remains consistent, whether your instance has a routed IP enabled or is a legacy one.

Reviewer 2: ```suggestion
  Your IP address will be retained during stop/start operations only if you have linked a flexible public IP address to your Instance. If you are using a dynamic IP address, the behavior remains consistent, whether your Instance has a routed IP enabled or is a legacy one.
```
